### PR TITLE
feat(api): add healthchecks to API

### PR DESCRIPTION
**Pull request by bacongobbler**
to be tested in tandem with https://github.com/deis/workflow-cli/pull/83

### CODE REVIEW

File: api/config.go [hunk 1]
```diff
@@ -12,14 +23,128 @@ type ConfigUnset struct {
...
-	Registry map[string]interface{} `json:"registry,omitempty"`
-	Created  string                 `json:"created,omitempty"`
-	Updated  string                 `json:"updated,omitempty"`
-	UUID     string                 `json:"uuid,omitempty"`
+	Owner        string                 `json:"owner,omitempty"`
+	App          string                 `json:"app,omitempty"`
+	Values       map[string]interface{} `json:"values,omitempty"`
+	Memory       map[string]interface{} `json:"memory,omitempty"`
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthchecks map[string]Healthcheck `json:"healthchecks,omitempty"`
```
Comment (kmala) [inline api/config.go]: should this be healthcheck instead of healthchecks as there can be only one healthcheck with different probes?

Comment (bacongobbler) [inline api/config.go]: no, there are two different healthchecks: liveness and readiness. You can have both configured for the same app, but each healthcheck can be of only one probe type (httpGet, exec, tcpSocket)

Comment (kmala) [inline api/config.go]: isn't healthcheck\s json going to be 
```
{
readinessProbe: {}
livenessProbe: {}
}
```
  and httpGet, exec, tcpSocket are just handlers in each in probe

File: api/config.go [hunk 2]
```diff
@@ -12,14 +24,128 @@ type ConfigUnset struct {
...
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthcheck  map[string]Healthcheck `json:"healthcheck,omitempty"`
+	Tags         map[string]interface{} `json:"tags,omitempty"`
+	Registry     map[string]interface{} `json:"registry,omitempty"`
+	Created      string                 `json:"created,omitempty"`
+	Updated      string                 `json:"updated,omitempty"`
+	UUID         string                 `json:"uuid,omitempty"`
+}
+
+type Healthcheck interface {
```
Comment (arschles) [inline api/config.go]: can you add some (simple) GoDoc for this and the below structs and funcs?

**Comment by arschles** [comment]
LGTM aside from my [docs comment](https://github.com/deis/controller-sdk-go/pull/10#discussion_r67786308)

File: api/config.go (hunk 2 above)
Comment (bacongobbler) [inline api/config.go]: certainly!

File: api/config.go [hunk 3]
```diff
@@ -12,14 +24,128 @@ type ConfigUnset struct {
...
+func (t TCPSocketProbe) String() string {
+	return fmt.Sprintf("Port: %s", t.Port)
+}
+
+type HealthcheckHTTPGetProbe struct {
+	InitialDelaySeconds string        `json:"initialDelaySeconds"`
+	TimeoutSeconds      string        `json:"timeoutSeconds"`
+	PeriodSeconds       string        `json:"periodSeconds"`
+	SuccessThreshold    string        `json:"successThreshold"`
+	FailureThreshold    string        `json:"failureThreshold"`
```
Comment (kmala) [inline api/config.go]: All these should be integers.
http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe

Comment (helgi) [inline api/config.go]: Indeed, the API validates them as int

Comment (bacongobbler) [inline api/config.go]: sounds good! I know that some values in the API are ints yet are validated as string. I'll change this around. :)

File: api/config.go [hunk 4]
```diff
@@ -12,14 +24,150 @@ type ConfigUnset struct {
...
+		msg += cmd + " "
+	}
+	return strings.TrimSpace(msg)
+}
+
+// HTTPGetProbe performs an HTTP GET request to the Pod
+// with the given path, port and headers.
+type HTTPGetProbe struct {
+	Path        string   `json:"path"`
+	Port        int   `json:"port"`
```
Comment (kmala) [inline api/config.go]: port can be a string 
http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_httpgetaction

Comment (helgi) [inline api/config.go]: It has to map to a Service port name (if it is a string) but we don't support that

Comment (bacongobbler) [inline api/config.go]: it is now a string.

**Pull request merged**